### ISSUE TITLE
encrypt() with an empty string salt does not work as expected

### ISSUE THREAD

**Issue by wanglin86769** [markusberg/unixcrypt#18]
Hello, thanks very much for developing the SHA256-based Unix crypt implementation for JavaScript, which is wonderful and surprising.

I am porting a Python program to JavaScript, and the Python program use passlib to encrypt user passwords, so I am looking for a JavaScript library that can generate the same result as the Python passlib, and found this unixcrypt library is very helpful.

Everything works fine except for the empty string salt, for example, the following statement cannot generate the same result as the Python passlib since empty string salt is not taken into account in unixcrypt,

JavaScript code

```
var unixcrypt = require("unixcrypt");
const pwHash = unixcrypt.encrypt("123456", "$5$rounds=5000$"); // Intended to pass in an empty string salt
```

Python code
```
from passlib.hash import sha256_crypt
sha256_crypt.using(salt='', rounds=5000).hash("123456")
```

If I use the generated JavaScript file in unixcrypt/dist directly and change the following line

```
if (parts[3]) {
        conf.saltString = parts[3];
}
```

to

```
if (parts[3] || parts[3] === '') {
        conf.saltString = parts[3];
}
```

it will generate the same result as Python passlib.

**Comment by markusberg** [comment]
Interesting, thank you for this. It certainly looks like a bug. Likely due to the api I chose. :-) 

I might need to publish a new major version.

**Comment by markusberg** [comment]
Alright, sorry for the delay. I just published version 1.2.0 which fixes this bug.

**Comment by wanglin86769** [comment]
Version 1.2.0 works fine, thank you.
